app/code/community/Bolt/Boltpay/controllers/ConfigurationController.php

```diff
@@ -0,0 +1,216 @@
+            $this->setErrorResponseData(
+                $responseData,
+                Mage::helper('boltpay')->__('Publishable Key - Multi-Page Checkout is invalid')
+            );
+        }
+        if (!($this->checkPublishableKeyOnePage())) {
+            $this->setErrorResponseData(
+                $responseData,
+                Mage::helper('boltpay')->__('Publishable Key - One Page Checkout is invalid')
+            );
+        }
+
+
+        // Validate database schema
+        if (!($this->checkSchema())) {
+            $this->setErrorResponseData(
+                $responseData,
+                Mage::helper('boltpay')->__('Schema is invalid')
+            );
+        }
+
+        $response = Mage::helper('core')->jsonEncode($responseData);
+        $this->getResponse()->setHeader('Content-type', 'application/json');
+        $this->getResponse()->setBody($response);
+    }
+
+
+    /**
+     * Transmit sign request to bolt to check whether api key is correct
+     * @return bool
+     */
+    protected function checkApiKey()
+    {
+        /** @var Bolt_Boltpay_Helper_Api $boltHelper */
+        $boltHelper = Mage::helper('boltpay/api');
+
+        $signRequest = array(
+            'merchant_user_id' => 'USER_ID_TEST_' . time(),
+        );
+
+        try {
+            $signResponse = $boltHelper->transmit('sign', $signRequest, 'merchant', 'merchant', $this->_storeId);
+        } catch (\Exception $e) {
+            return false;
+        }
+
+        if (!$signResponse) {
+            return false;
+        }
+
+        return true;
+    }
+
+
+    /**
+     * @return bool
+     */
+    protected function checkSigningSecret()
+    {
+        // Currently there isn't a way to validate the signing secret since this must come from Bolt.
+        return true;
+    }
+
+    /**
+     * @return bool
+     */
+    protected function checkPublishableKeyMultiPage()
+    {
+        $keyMultiplePage = Mage::getStoreConfig('payment/boltpay/publishable_key_multipage', $this->_storeId);
+
+        return !$keyMultiplePage || $this->curlCheckPublishableKey($keyMultiplePage);
+    }
+
+    /**
+     * Calls the Bolt API endpoint.
+     *
+     * @param $key
+     * @return mixed thrown if an error is detected in a response
+     */
+    protected function curlCheckPublishableKey($key)
+    {
+        /** @var Bolt_Boltpay_Helper_Api $boltHelper */
+        $boltHelper = Mage::helper('boltpay/api');
+
+        $url = $boltHelper->getApiUrl($this->_storeId) . 'v1/merchant';
+
+        $ch = curl_init($url);
+
+        $headerInfo = array(
+            "X-Publishable-Key: $key"
+        );
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headerInfo);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_HEADER, true);
+
+        curl_exec($ch);
+        $response = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+
+        return $response == 200;
```

Comment:
can you change this to something like int($response/100) == 2?
We may use different response code like 201 created in the future